### PR TITLE
Scan function refactoring

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/main/kotlin/Scanner.kt [hunk 1]
```diff
@@ -109,41 +109,28 @@ class Scanner(
...
-            scan(
-                packages,
-                ScanContext(
-                    ortResult.labels,
-                    PackageType.PACKAGE,
-                    ortResult.repository.config.excludes,
-                    scannerConfig.detectedLicenseMapping
-                )
+        val packages = ortResult.getPackages(skipExcluded).map { it.metadata }.filterNotConcluded()
+            .filterNotMetadataOnly().toSet()
```
Comment (fviernau) [inline scanner/src/main/kotlin/Scanner.kt:125]: Maybe add an empty line here.

File: scanner/src/main/kotlin/Scanner.kt [hunk 2]
```diff
@@ -109,41 +109,28 @@ class Scanner(
...
+        val packageResults = scan(
+            packages,
+            ScanContext(
+                ortResult.labels,
```
Comment (fviernau) [inline scanner/src/main/kotlin/Scanner.kt:129]: out of scope: Is it correct that line 128 is different from line 116?

**Review by fviernau — APPROVED** [review]

File: scanner/src/main/kotlin/Scanner.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/Scanner.kt:125]: I actually had it there first, but then it looked a bit odd compared to the equivalent block with lines 113 / 114, so I decided to not add an empty line here to keep more compact blocks of code. But I don't care that much, I could add an empty line here (and also after line 113) if you prefer.

Comment (fviernau) [reply scanner/src/main/kotlin/Scanner.kt:125]: let's leave it as-is.

File: scanner/src/main/kotlin/Scanner.kt (hunk 2 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/Scanner.kt:129]: Good question. I suspect this should also say `ortResult.labels + labels`, @mnonnenmacher?

**Pull request merged**